src/main/java/access/Access.java

```diff
@@ -110,65 +106,38 @@ public void setIsChapterLevel() {
     }
 
     public void setModuleLevel(String moduleLevel) throws IncorrectAccessLevelException {
-        if (isChapterLevel) {
-            throw new IncorrectAccessLevelException("Sorry, you currently are in the chapter level.");
-        }
-
-        if (isModuleLevel) {
-            if (!(moduleLevel.equals(""))) {
-                throw new IncorrectAccessLevelException("Sorry, you are already in the module level, "
-                        + "please go back to admin level first.");
-            }
-
-            String replacement = "/" + this.moduleLevel;
-            this.level = level.replace(replacement, "");
-            this.moduleLevel = moduleLevel;
-            this.module = null;
-            this.isModuleLevel = false;
-            this.isAdminLevel = true;
-            return;
-        }
-
         if (isAdminLevel) {
-            if (moduleLevel.equals("")) {
-                throw new IncorrectAccessLevelException("Sorry, you are already in the admin level.");
-            }
             this.moduleLevel = moduleLevel;
             this.level = level + "/" + moduleLevel;
             this.module = new Module(moduleLevel);
             this.isModuleLevel = true;
             this.isAdminLevel = false;
```

Comment:
you can use setIsModuleLevel() to do this

Reply:
Thank you, I have changed all the separated code into setIsModuleLevel().